libr/arch/p/bf/bf.c

```diff
@@ -178,12 +196,12 @@ static int bf_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *_buf, int len,
 					op->type = R_ANAL_OP_TYPE_ILL;
 					goto beach;
 				}
-				if (i == len - 1 && anal->read_at) {
+				if (i == len - 1) {
 					int new_buf_len = len + 1 + BUFSIZE_INC;
 					ut8 *new_buf = calloc (new_buf_len, 1);
 					if (new_buf) {
 						free (buf);
-						(void)anal->read_at (anal, addr, new_buf, new_buf_len);
+						(void)read_at (bin->iob.io, addr, new_buf, new_buf_len);
```

Comment:
ideally rarch shouldnt be directly accessing io, and should use only the data passed in op->bytes